main/src/ca/uwaterloo/flix/util/ParOps.scala

```diff
@@ -90,6 +91,16 @@ object ParOps {
       case (k, v) => (k, f(v))
     }.toMap
 
+  /**
+    * Applies the function `f` to every value of the map `m` in parallel.
+    */
+  def parMapValueList[K, A, B](m: ListMap[K, A])(f: List[A] => List[B])(implicit flix: Flix): ListMap[K, B] =
```

Comment:
I think it's more straightforward if this function f is just A => B. I am guessing that the functions called `visitInstanceList` just map over the list anyway, so we can get rid of those.